### PR TITLE
docs: fix and improve docstrings

### PR DESCRIPTION
**Pull request by rouille**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Fix and improve docstrings

### Time estimate
10 min

### CODE REVIEW

File: powersimdata/design/investment/create_mapping_files.py [hunk 1]
```diff
@@ -8,25 +8,23 @@
...
+    """Perform a spatial join between two input layers.
+
     :param geopandas.GeoDataFrame left_df: A dataframe of Points.
-    :param geopandas.GeoDataFrame right_df: A dataframe of Polygons/Multipolygons
-    :param float/int search_dist: parameter (specified in map units) is used to limit
-        the search area for geometries around source points. Smaller -> faster runtime.
-    :return: (*geopandas.GeoDataFrame*) -- A dataframe of Points mapped to each polygon
-        in right_df.
+    :param geopandas.GeoDataFrame right_df: A dataframe of Polygons/Multipolygons.
+    :param float/int search_dist: radius (in map umits) around point to detect polygons.
```
Comment (danielolsen) [inline powersimdata/design/investment/create_mapping_files.py:15]: `umits` -> `units`

File: powersimdata/design/investment/create_mapping_files.py [hunk 2]
```diff
@@ -156,18 +150,14 @@ def bus_to_reeds_reg(df):
...
-
-    Note: mapping may take a while, especially for many points.
+    """Map bus to ReEDS regions.
+
+    :param pandas.DataFrame df: bus data frame.
+    :return: (*pandas.DataFrame*) -- index: bus id, columns: lat, lon, name_abbr
+        (NEEM region)
+
+    .. note: the shapefile used for mapping is pulled from the Energy Zones `Mapping
+        tool <http://ezmt.anl.gov>`_. This map is overly detailed, so the shapes are simplified using 1 km distance (Douglas-Peucker) method in QGIS.
```
Comment (danielolsen) [inline powersimdata/design/investment/create_mapping_files.py:160]: Line too long?

File: powersimdata/design/investment/investment_costs.py [hunk 3]
```diff
@@ -135,8 +130,8 @@ def get_transformer_mult(x, bus_reg, ac_reg_mult, xfmr_lookup_alerted=set()):
     lines[["MW", "costMWmi"]] = lines.apply(lambda x: select_mw(x, ac_cost), axis=1)
 
     # check that all buses included in this file and lat/long values match,
-    #   otherwise re-run mapping script on mis-matching buses.
-    # these buses are missing in region file
+    # otherwise re-run mapping script on mis-matching buses. Tthese buses are missing
```
Comment (danielolsen) [inline powersimdata/design/investment/investment_costs.py:133]: `Tthese` -> `These`

File: powersimdata/design/investment/create_mapping_files.py (hunk 2 above)
Comment (BainanXia) [reply powersimdata/design/investment/create_mapping_files.py:160]: I don't think black will apply line length filter on doc strings. I used to leave it as it is or manually fix it.

File: powersimdata/design/investment/investment_costs.py [hunk 4]
```diff
@@ -275,20 +267,19 @@ def _calculate_single_line_cost(line, bus):
...
 
 def calculate_gen_inv_costs(scenario, year, cost_case, sum_results=True):
-    """Given a Scenario object, calculate the total cost of building that scenario's
-        upgrades of generation.
-    Currently only uses one (arbutrary) sub-technology. Drops the rest of the costs.
-        Will want to fix for wind/solar (based on resource supply curves).
-    Currently uses ReEDS regions to find regional multipliers.
+    """Calculate cost of upgrading generators in a scenario. ReEDS regions are used to
+    find regional multipliers.
+    upgrades of generation.
```
Comment (danielolsen) [inline powersimdata/design/investment/investment_costs.py:272]: Should these be part of the same line?

File: powersimdata/design/investment/investment_costs.py [hunk 5]
```diff
@@ -275,20 +267,19 @@ def _calculate_single_line_cost(line, bus):
...
-        'Advanced': generally lower costs
-    :return: (*pandas.DataFrame*) -- Total generation investment cost summed by
+    :param int/str year: building year.
+    :param str cost_case: ATB cost case of data. *'Moderate'*: mid cost case,
+        *'Conservative'*: generally higher costs, *'Advanced'*: generally lower costs
+    :return: (*pandas.DataFrame*) -- total generation investment cost summed by
         technology.
+
+    .. todo:: it currently uses one (arbitrary) sub-technology. The rest of the costs
+        are dropped. Wind and solar will need to be fixed based on theresource supply curves.
```
Comment (danielolsen) [inline powersimdata/design/investment/investment_costs.py:282]: `theresource` -> `the resource`

File: powersimdata/design/investment/investment_costs.py [hunk 6]
```diff
@@ -317,38 +308,32 @@ def calculate_gen_inv_costs(scenario, year, cost_case, sum_results=True):
...
     :raises ValueError: if year not 2020 - 2050, or cost case not an allowed option.
-    :raises TypeError: if year gets the wrong type, or if cost_case is not str.
-    :return: (*pandas.Series*) -- Total generation investment cost,
-        summed by technology.
+    :raises TypeError: if year not int/str or cost_case not str.
+    :return: (*pandas.Series*) -- total generation investment cost, summed by
+        technology.
+
+    .. note:: the function computes the total capital cost as:
+        CAPEX_total = CAPEX ($/MW) * Pmax (MW) * multiplier
```
Comment (danielolsen) [inline powersimdata/design/investment/investment_costs.py:324]: `multiplier` -> `regional multiplier`

File: powersimdata/design/investment/create_mapping_files.py (hunk 1 above)
Comment (rouille) [reply powersimdata/design/investment/create_mapping_files.py]: done

File: powersimdata/design/investment/create_mapping_files.py (hunk 2 above)
Comment (rouille) [reply powersimdata/design/investment/create_mapping_files.py]: fixed

File: powersimdata/design/investment/investment_costs.py (hunk 3 above)
Comment (rouille) [reply powersimdata/design/investment/investment_costs.py]: done

File: powersimdata/design/investment/investment_costs.py (hunk 4 above)
Comment (rouille) [reply powersimdata/design/investment/investment_costs.py]: last lines was deleted

File: powersimdata/design/investment/investment_costs.py (hunk 5 above)
Comment (rouille) [reply powersimdata/design/investment/investment_costs.py]: done

File: powersimdata/design/investment/investment_costs.py (hunk 6 above)
Comment (rouille) [reply powersimdata/design/investment/investment_costs.py]: done

**Review by danielolsen — APPROVED** [review]
Thanks

**Pull request merged**